tests/compatibility_tests/create_structured_index/test_create_structured_index.py

```diff
@@ -0,0 +1,66 @@
+
+@pytest.mark.marqo_version('2.0.0')
+class TestCreateStructuredIndex(BaseCompatibilityTestCase):
+    indexes_settings_to_test_on = [
+        {
+            "type": "structured",
+            "vectorNumericType": "float",
+            "model": "open_clip/ViT-B-32/laion2b_s34b_b79k",
+            "normalizeEmbeddings": True,
+            "textPreprocessing": {
+                "splitLength": 2,
+                "splitOverlap": 0,
+                "splitMethod": "sentence",
+            },
+            "imagePreprocessing": {"patchMethod": None},
+            "allFields": [
+                {"name": "text_field", "type": "text", "features": ["lexical_search"]},
+                {"name": "caption", "type": "text", "features": ["lexical_search", "filter"]},
+                {"name": "tags", "type": "array<text>", "features": ["filter"]},
+                {"name": "image_field", "type": "image_pointer"},
+                {"name": "my_int", "type": "int", "features": ["score_modifier"]},
+                # this field maps the above image field and text fields into a multimodal combination.
+                {
+                    "name": "multimodal_field",
+                    "type": "multimodal_combination",
+                    "dependentFields": {"image_field": 0.9, "text_field": 0.1},
+                },
+            ],
+            "tensorFields": ["multimodal_field"],
+            "annParameters": {
+                "spaceType": "prenormalized-angular",
+                "parameters": {"efConstruction": 512, "m": 16},
+            }
+        }]
+    indexes_to_test_on = ["test_create_index_api_structured_index"]
+
+    @classmethod
+    def tearDownClass(cls) -> None:
+        cls.indexes_to_delete = cls.indexes_to_test_on
+        super().tearDownClass()
+
+    @classmethod
+    def setUpClass(cls) -> None:
+        super().setUpClass()
+
+    def prepare(self):
+        self.logger.info(f"Creating indexes {self.indexes_settings_to_test_on}")
+        all_results = {}
+        for index_name, index_settings in zip(self.indexes_to_test_on, self.indexes_settings_to_test_on):
+            self.client.create_index(index_name, settings_dict = index_settings)
+            all_results[index_name] = self.client.index(index_name).get_settings()
+        self.save_results_to_file(all_results)
+
+    def test_expected_settings(self):
+        expected_settings = self.load_results_from_file()
+        for index_name in self.indexes_to_test_on:
+            try:
+                actual_settings = self.client.index(index_name).get_settings()
+                self.logger.debug(f"Printing actual_settings {actual_settings}")
+                self.logger.debug(f"Printing expected_settings {expected_settings.get(index_name)}")
+            except Exception as e:
+                raise Exception(f"Exception when getting index settings for index {index_name}") from e
```

Comment:
This is a bit odd. We're not expecting an error here but if one does happen the test fails. I don't think you need try/except at all

Reply:
try except is required so that even if one test fails the other can run. The issue here was that I was raising another exception anyway. I have changed that and now I let the other tests run before reporting exceptions